File: examples/python/in_code.py
#!/usr/bin/env python3
"""
Specado Python Example: Define Prompts in Code

Define a prompt specification directly in your code.
"""

import json
from specado import Client


def main():
    # Initialize the client with a friendly provider name and custom model
    client = Client("openai", model="gpt-5")

    print("Executing in-code prompt specification...")
    print("-" * 60)

    response = client.complete_text(
        "Explain what a closure is in JavaScript in one paragraph.",
        system_message="You are a helpful assistant that explains programming concepts clearly.",
        temperature=0.5,
    )

    # Display the results
    print(json.dumps(response, indent=2))


if __name__ == "__main__":
    main()

File: examples/node/in_code.js
#!/usr/bin/env node
/**
 * Specado Node.js Example: Define Prompts in Code
 */

const { Client } = require('specado');

async function main() {
  const client = new Client('openai', { model: 'gpt-5' });

  console.log('Executing in-code prompt specification...');
  console.log('-'.repeat(60));

  const response = await client.completeText(
    'Explain what a closure is in JavaScript in one paragraph.',
    {
      system: 'You are a helpful assistant that explains programming concepts clearly.',
      temperature: 0.5,
    }
  );

  console.log(JSON.stringify(response, null, 2));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
